connector/connect/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -62,3 +65,29 @@ message CreateScalarFunction {
     FUNCTION_LANGUAGE_SCALA = 3;
   }
 }
+
+// As writes are not directly handled during analysis and planning, they are modeled as commands.
+message WriteOperation {
+  Relation input = 1;
+  string format = 2;
+
+  oneof save_type {
+    string path = 3;
+    string table_name = 4;
+  }
+  string mode = 5;
+  repeated string sortColumnNames = 6;
+  repeated string partitionByColumns = 7;
+  BucketBy bucketBy = 8;
+  repeated WriteOptions options = 9;
```

Comment:
Why just not a map<string, string>? Does proto map have a iterator (if so in later use it is the same as `repeated`)

Reply:
So the interesting question is, options at least in the Scala source supports different types. It should probably be a `map<string, expressions.Literal>` but I was too lazy. I can change it.

Reply:
I mapped this to `map<string, string>` because behind this scenes it's converted to a string anyways.